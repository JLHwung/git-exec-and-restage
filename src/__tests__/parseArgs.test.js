import { parseArgs } from "../";

describe("parseArgs", () => {
  it("empty", () => {
    expect(parseArgs([])).toMatchSnapshot();
  });
  it("command only", () => {
    expect(parseArgs(["prettier"])).toMatchSnapshot();
  });
  it("command and 1 file", () => {
    expect(parseArgs(["prettier", "a.js"])).toMatchSnapshot();
  });
  it("command and 2 files", () => {
    expect(parseArgs(["prettier", "a.js", "b.js"])).toMatchSnapshot();
  });
  it("command, separator and 1 file", () => {
    expect(parseArgs(["prettier", "--", "a.js"])).toMatchSnapshot();
  });
  it("command, 1 option and 1 file", () => {
    expect(parseArgs(["prettier", "--write", "--", "a.js"])).toMatchSnapshot();
  });
  it("command, 1 option, two separators, 1 file", () => {
    expect(
      parseArgs(["prettier", "--write", "--", "--", "a.js"])
    ).toMatchSnapshot();
  });
});
