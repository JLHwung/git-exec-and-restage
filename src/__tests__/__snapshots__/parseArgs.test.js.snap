// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseArgs command and 1 file 1`] = `
Object {
  "command": "prettier",
  "commandArgs": Array [],
  "files": Array [
    "a.js",
  ],
}
`;

exports[`parseArgs command and 2 files 1`] = `
Object {
  "command": "prettier",
  "commandArgs": Array [],
  "files": Array [
    "a.js",
    "b.js",
  ],
}
`;

exports[`parseArgs command only 1`] = `
Object {
  "command": "prettier",
  "commandArgs": Array [],
  "files": Array [],
}
`;

exports[`parseArgs command, 1 option and 1 file 1`] = `
Object {
  "command": "prettier",
  "commandArgs": Array [
    "--write",
  ],
  "files": Array [
    "a.js",
  ],
}
`;

exports[`parseArgs command, 1 option, two separators, 1 file 1`] = `
Object {
  "command": "prettier",
  "commandArgs": Array [
    "--write",
  ],
  "files": Array [
    "a.js",
  ],
}
`;

exports[`parseArgs command, separator and 1 file 1`] = `
Object {
  "command": "prettier",
  "commandArgs": Array [],
  "files": Array [
    "a.js",
  ],
}
`;

exports[`parseArgs empty 1`] = `
Object {
  "command": undefined,
  "commandArgs": Array [],
  "files": Array [],
}
`;
